src/Value/AppId.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * Copyright 2015 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupU2fBundle\Value;
+
+use Surfnet\StepupU2fBundle\Exception\InvalidArgumentException;
+
+final class AppId
+{
+    /**
+     * @var string
+     */
+    private $appId;
+
+    /**
+     * @param string $appId
+     */
+    public function __construct($appId)
+    {
+        if (!is_string($appId)) {
```

Comment:
check, deviates a bit from how we've done it in other places, but makes sense here.